app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -2206,12 +2206,19 @@ private void moveFocusToMainFragment(final boolean toMain) {
             mainFragment.setDescendantFocusability(afterDescendants);
             toolbar.setDescendantFocusability(afterDescendants);
             ((ViewGroup) requireView()).setDescendantFocusability(blockDescendants);
-            mainFragment.requestFocus();
+            // Check if the mainFragment (search-results) or the toolbar (Textfield for search)
+            // have focus. If so then don't focus them again.
+            if (!mainFragment.hasFocus() && !toolbar.hasFocus()) {
+                mainFragment.requestFocus();
+            }
```

Comment:
I find that these types of comments are more confusing than helpful, since they are simply describing what the following line of code does. In this case it's just a simple if expression.
Here I was confused because the comment is basically an inversion of the whole expression.
``` java
if (mainFragment.hasFocus() || toolbar.hasFocus()) {
    // do nothing
} else {
    mainFragment.requestFocus();
}
```
Same for the one below

---

Adding the reason for that line (this specific bug) on the other hand would help a lot in the case that it ever can be removed.

Reply:
Thank you for the hint.
Should be fixed now: Made the comments easier to understand.

---

> Adding the reason for that line (this specific bug) on the other hand would help a lot in the case that it ever can be removed.

I hope this should also be fixed now with the new comments.
In case of emergency people can also use ``git blame`` :)